projects/components/src/file-upload/drop-zone/drop-zone.directive.ts

```diff
@@ -0,0 +1,31 @@
+import { Directive, EventEmitter, HostListener, Output } from '@angular/core';
+
+@Directive({
+  selector: '[htDropZone]'
+})
+export class DropZoneDirective {
+  @Output()
+  public readonly dropped: EventEmitter<FileList> = new EventEmitter();
+
+  @Output()
+  public readonly dragOver: EventEmitter<boolean> = new EventEmitter();
+
+  @HostListener('drop', ['$event'])
+  public onDrop(event: DragEvent): void {
+    event.preventDefault();
+    this.dropped.emit(event.dataTransfer?.files);
+    this.dragOver.emit(false);
+  }
+
+  @HostListener('dragover', ['$event'])
+  public onDragover(event: DragEvent): void {
+    event.preventDefault();
+    this.dragOver.emit(true);
+  }
+
+  @HostListener('dragleave', ['$event'])
+  public onDragleave(event: DragEvent): void {
+    event.preventDefault();
+    this.dragOver.emit(false);
```

Comment:
Probably, we should add both dragOver and dragLeave as a separate output. It is not clear that dragOver: false is same as dragLeave

Reply:
ping. Also, isn't `DragLeave` the same as dropped? Do we need both?

Reply:
So `mouseover` and `mouseleave` are combined as `hover`. So I can combine it as `dragHover` output.
No `DragLeave` is different than `dropped`, in drop we're dropping the files but in the drag leave we're moving away without dropping.